Source/JavaScriptCore/heap/GCSegmentedArrayInlines.h

```diff
@@ -180,7 +180,7 @@ inline void GCSegmentedArray<T>::validatePrevious()
     unsigned count = 0;
     for (GCArraySegment<T>* current = m_segments.head(); current; current = current->next())
         count++;
-    ASSERT(m_segments.size() == m_numberOfSegments);
+    ASSERT_UNUSED(count, m_segments.size() == m_numberOfSegments);
```

Comment:
This makes no sense. Why are we computing `count` and not using it? I think the code above to compute count is dead code. If we are computing `count` then it seems we should be asserting it is `== m_segments.size()`?

Reply:
I'm assuming there's some kind of validation happening here but maybe someone on the @WebKit/jsc-reviewers side can verify

Reply:
I think the old ASSERT looks wrong. We should do `ASSERT(count == m_numberOfSegments)` probably here.

Reply:
Or, maybe, we should skip count. `m_segments.size()` would be doing the same thing (let's check that).

Reply:
updating to `count == m_numberOfSegments`

Reply:
> Or, maybe, we should skip count. `m_segments.size()` would be doing the same thing (let's check that).

Yes that’s what I was thinking too. Someone had the idea of using `m_segments.size()` but forgot to delete the code that does the same thing.